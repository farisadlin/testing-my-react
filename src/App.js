import {createContext, useState} from 'react'

import "./App.css";
import MyAxios from "./components/MyAxios";
import MyContext from './components/MyContext';
import { MyForm } from "./components/MyForm";


export const CountContext = createContext()

function App() {
  const [count, setCount] = useState(0)

  return (
    <CountContext.Provider value={{count, setCount}}>
      <MyForm />
      <MyAxios />
      <MyContext />
    </CountContext.Provider>
  );
}

export default App;
