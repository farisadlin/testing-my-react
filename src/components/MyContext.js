import React, { useContext } from 'react'
import {CountContext} from '../App'

const MyContext = () => {
    const {count, setCount} = useContext(CountContext)
    const increment = () => {
        setCount(count + 1)
    }

    const decrement = () => {
        setCount(count - 1)
    }
    return (
        <>
        <h1 style={{textAlign: "center"}}>This is testing for React Context</h1>
        <div style={{display: "flex", justifyContent: "space-around", margin:"50px 0"}}>
            <button onClick={decrement}>-</button>
            <span>{count}</span>
            <button onClick={increment}>+</button>
        </div>
        </>
    )
}

export default MyContext
