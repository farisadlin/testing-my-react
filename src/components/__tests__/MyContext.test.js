import React from "react";
import { render, screen, waitFor, cleanup, act } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import MyContext from "../MyContext";
import { CountContext } from "../../App";

let renderContext;

const { count, setCount } = jest.fn();

describe("Testing react context", () => {
  beforeAll(() => {
    renderContext = render(
      <CountContext.Provider value={{ count, setCount }}>
        <MyContext />
      </CountContext.Provider>
    );
  });

  afterEach(cleanup);

  it("detect context in component", () => {
    const tree = renderer
      .create(
        <CountContext.Provider value={{ count, setCount }}>
          <MyContext />
        </CountContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
