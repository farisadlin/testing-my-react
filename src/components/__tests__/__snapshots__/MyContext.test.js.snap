// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react context detect context in component 1`] = `
Array [
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    This is testing for React Context
  </h1>,
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-around",
        "margin": "50px 0",
      }
    }
  >
    <button
      onClick={[Function]}
    >
      -
    </button>
    <span />
    <button
      onClick={[Function]}
    >
      +
    </button>
  </div>,
]
`;
